submittal/src/main/java/com/kyriba/submittal/service/TopicServiceImpl.java

```diff
@@ -0,0 +1,51 @@
+package com.kyriba.submittal.service;
+
+import com.kyriba.submittal.dao.TopicRepository;
+import com.kyriba.submittal.domain.Topic;
+import com.kyriba.submittal.domain.TopicStatus;
+import com.kyriba.submittal.service.exception.ObjectNotFoundException;
+import com.kyriba.submittal.service.exception.UnsupportedTopicTransitionException;
+import lombok.RequiredArgsConstructor;
+import org.springframework.lang.NonNull;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+
+/**
+ * @author M-ABL
+ */
+@Service
+@Transactional
+@RequiredArgsConstructor
+public class TopicServiceImpl implements TopicService
+{
+  private final TopicRepository topicRepository;
+
+
+  @Override
+  public void transit(final long topicId, @NonNull final TopicStatus status)
+  {
+    Topic topic = findTopicById(topicId);
+    if (topic.getStatus() == TopicStatus.DRAFT) {
+      throw new ObjectNotFoundException("The requested topic is not ready");

```

Comment:
ObjectNotFoundException seems strange for this case. In fact, object is found, but its state is illegal

Reply:
business logic considers such a topic as non-existent for the end user. In a good case this request must not be executed at all